app/controllers/MovementDetailsController.scala

```diff
@@ -72,24 +75,26 @@ class MovementDetailsController @Inject()(
   }
 
   private def handleSavingArrival(arrivalAnswers: ArrivalAnswers)(implicit request: JourneyRequest[AnyContent]): Future[Either[Html, Call]] = {
-    val boundForm = details.arrivalForm().bindFromRequest()
-
-    boundForm.fold(
-      (formWithErrors: Form[ArrivalDetails]) => Future.successful(Left(arrivalDetailsPage(formWithErrors, arrivalAnswers.consignmentReferences))),
-      validForm =>
-        cache.upsert(request.cache.update(arrivalAnswers.copy(arrivalDetails = Some(validForm)))).map { _ =>
-          Right(controllers.routes.LocationController.displayPage())
-      }
-    )
+    val consignmentReferenceValue = arrivalAnswers.consignmentReferences.map(_.referenceValue).getOrElse(throw ReturnToStartException)
+    details
+      .arrivalForm()
+      .bindFromRequest()
+      .fold(
+        (formWithErrors: Form[ArrivalDetails]) => Future.successful(Left(arrivalDetailsPage(formWithErrors, consignmentReferenceValue))),
+        validForm =>
+          cache.upsert(request.cache.update(arrivalAnswers.copy(arrivalDetails = Some(validForm)))).map { _ =>
+            Right(controllers.routes.LocationController.displayPage())
+        }
+      )
   }
 
   private def handleSavingDeparture(departureAnswers: DepartureAnswers)(implicit request: JourneyRequest[AnyContent]): Future[Either[Html, Call]] = {
-    def consignmentReference = departureAnswers.consignmentReferences.map(_.referenceValue).getOrElse(throw ReturnToStartException)
+    val consignmentReferenceValue = departureAnswers.consignmentReferences.map(_.referenceValue).getOrElse(throw ReturnToStartException)
```

Comment:
why rename to `value` ?

Reply:
`ConsignmentReference` is a case class, but the value under this field is of type `String`. I was confused when started working in this area, so decided that it is better to explicitly state that this is not an instance of `ConsignmentReference`.
I thought about calling it just `ucr`, because essentially it is a UCR. But adding "Value" as a suffix introduces less changes.